repositories/suite_test.go

```diff
@@ -0,0 +1,70 @@
+package repositories_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/api/v1alpha1"
+	. "github.com/onsi/gomega"
+	"github.com/sclevine/spec"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var (
+	suite             spec.Suite
+	testEnv           *envtest.Environment
+	k8sClient         client.Client
+	k8sConfig         *rest.Config
+	testServerAddress string
+)
+
+func Suite() spec.Suite {
+	if suite == nil {
+		suite = spec.New("API Shim")
+	}
+
+	return suite
+}
+
+func SuiteDescribe(desc string, f func(t *testing.T, when spec.G, it spec.S)) bool {
+	return Suite()(desc, f)
+}
+
+func TestSuite(t *testing.T) {
+	Suite().Before(func(t *testing.T) {
```

Comment:
We have adopted a different pattern for initializing envtest. See [webhook suite setup](https://github.com/cloudfoundry/cf-k8s-controllers/blob/588dfa53039b9f496a5724250a4c24dc1d698ae6/webhooks/suite_integration_test.go#L45-L52).
 
**Background** - Suite.Before() and After() functions were being reevaluated for every `spec`, this was time consuming with setting up and tearing down envtest. 

The new pattern allows you to setup once and run all our tests.

Reply:
https://github.com/cloudfoundry/cf-k8s-api/pull/30/commits/85984449de62e66ae67fa7256dc79d99cc3aba9f